src/main/java/com/instaclustr/cassandra/driver/auth/KerberosAuthProvider.java

```diff
@@ -222,18 +239,25 @@ public Authenticator newAuthenticator(InetSocketAddress host, String authenticat
         private final Subject subject;
         private final SaslClient saslClient;
 
-        private KerberosAuthenticator(String authorizationId, String saslProtocol, InetSocketAddress host, Map<String, ?> saslProperties)
+        private KerberosAuthenticator(String authorizationId, String saslProtocol, EndPoint endpoint, Map<String, ?> saslProperties)
         {
-            Preconditions.checkNotNull(saslProtocol);
```

Comment:
Is it valid to drop this? I'd suggest it should just not be using the transitive checkNotNull.

```
    if (saslProtocol == null) {
        throw new NullPointerException("No SASL protocol supplied, unable to perform kerberos authentication");
    }
```